tchannel/peer_heap.py

```diff
@@ -67,7 +84,28 @@ def pop_peer(self):
         return heap.pop(self)
 
     def push_peer(self, peer):
-        """Push a new peer into the heap"""
+        """Push a new peer into the heap
+
+        Order is equal to peer heap's current order plus random value within
+        the current peer size. It solves two problems when peers are in
+        same score:
+
+        1. 'dead peer' caused by random problem. If use fully random value, and
+        one peer is very unlock to get the worst value, as the peer selection
+        keeps working, that peer may not be selected anymore.
+
+        2. deterministic problem. If we just use push order, then all the
+        tchannel instance will follow the same pattern. If they all get the
+        same list of hyperbahn nodes, the peers that they pick to talk to
+        hyperbahn will be same. In that situation, it will cause the hyperbahn
+        node received very imbalance requests from tchannel clients.
```

Comment:
Change to:

> - Dead peers: If the order is completely random, then an unlucky peer with a very bad assigned order may never get selected.
> - Determinism: If the insertion order is used as-is, then all TChannel instances would follow the same selection pattern, causing load imbalance. For example if they get the same static list of peers, they will all pick the first one, then the second one, and so on, cycling between which host gets overloaded with requests.

Also, same concern as before: This behavioral description (and how PeerHeap solves it with order) belongs in the class docstring, not the method docstring. Like,

    class MyClass(object):
        """MyClass does X.

        It does Z to solve the following problems:
        - ...
        """